vortex-scan/src/multi_thread.rs

```diff
@@ -74,12 +74,18 @@ impl ScanBuilder<ArrayRef> {
             //  head-room to run ahead and figure out the I/O demands of subsequent tasks.
             .buffered(num_workers * concurrency);
 
-        Ok(
-            iter::from_fn(move || block_on(stream.next())).filter_map(|result| {
-                result
-                    .map_err(|e| vortex_err!("Failed to join on a spawned scan task {e}"))
-                    .vortex_expect("Failed to join on a spawned scan task")
-            }),
-        )
+        Ok(iter::from_fn(move || {
+            // Use runtime-aware blocking strategy
+            if tokio::runtime::Handle::try_current().is_ok() {
+                tokio::task::block_in_place(|| CPU_RUNTIME.handle().block_on(stream.next()))
+            } else {
+                block_on(stream.next())
```

Comment:
Isn't the bug that this should always be CPU_RUNTIME.block_on and not futures::executor::block_on? 

It seems even weird to use whatever the current tokio handle is?